pybossa.js

```diff
@@ -166,8 +167,122 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+            return { msg: 'You must complete a quiz successfully before you can work on this job. You are in Quiz mode.',
+                     type: 'info' }
+        }
+    }
+
+    // This func determine if there are no more questions available for the quiz.
+    function _outOfGoldenTasks (userProgress, quiz, config){
+        if (quiz && config.enabled && userProgress.remaining_for_user === 0 && quiz.status === 'in_progress' && !window.pybossa.isGoldMode){
+            return { msg: 'We have run out of quiz questions for you. Please notify the project owner.',
+                     type: 'info' }
+        }
+    }
+
+    //This func is to determine if in Bulk gold mode are more task available to be converted into gold
+    function _outOfNonGoldTask (userProgress){
+        if (userProgress.available_gold_tasks === userProgress.total && window.pybossa.isBulk){
+            return { msg:'In gold mode, there are no tasks available.',
+                     type: 'warning' };
+        }
+    }
+
+    //This func determine if the quiz is failed and the user is not able to work on the project
+    function _failedQuiz (quiz, config){
+        if (quiz && config.enabled && quiz.status === 'failed'){
+            return { msg: 'Thank you for taking the quiz. You got ' + quiz.result.right + ' correct out of ' + quiz.config.questions +' tasks. ' +
+                            'You have been blocked from working on this job. The administrator of this job will contact you with next steps.',
+                     type: 'error' };
+        }
+    }
+
+    // This func determine if the user passed the quiz and use two flags to decide to show or not the notification since we only want to show it once
+    function _passedQuiz (quiz, config){
+        if (quiz && config.enabled && quiz.status === 'passed' && (window.pybossa.passedQuizShowed || window.pybossa.takingQuiz)){
+            window.pybossa.passedQuizShowed = false;
+            window.pybossa.takingQuiz = false;
+            return { msg: 'Thank you for taking the quiz. You got ' + quiz.result.right + ' correct out of ' + quiz.config.questions + ' tasks. ' +
+                           'You will now be able to work on this job.',
+                     type: 'warning' };
+        }
+    }
+
+    //This func determine if the project is completed.
+    function _projectCompleted (userProgress, quiz){
+        if (quiz && userProgress.remaining_for_user === 0 && quiz.status !== 'in_progress'){
+            return { msg: 'Congratulations, you have completed the job.',
+                     type: 'success' };
+        }
+    }
+
+    //This func determine if its in gold mode.
+    function _inGoldMode (userProgress){
+        if ((window.pybossa.isGoldMode && !(userProgress.available_gold_tasks === userProgress.remaining)) ||
+            (window.pybossa.isGoldMode && !window.pybossa.isBulk) ){
+            return { msg: 'In Gold Mode', type: 'warning' };
```

Comment:
Your code is equivalent to this:
``` javaScript
if (!window.pybossa.isGoldMode) return;
if ((userProgress.available_gold_tasks === userProgress.remaining) && window.pybossa.isBulk) return;
return {  msg: 'In Gold Mode', type: 'warning' };
```
It's still 3 lines but it's a little easier to understand.